modules/siddhi-core/src/main/java/io/siddhi/core/util/extension/validator/InputParameterValidator.java

```
@@ -0,0 +1,134 @@
+        }
+
+        //Find the parameterOverLoad
+        ParameterOverload parameterOverload = null;
+        for (ParameterOverload aParameterOverload : parameterOverloads) {
+            String[] overloadParameterNames = aParameterOverload.parameterNames();
+            if (overloadParameterNames.length == attributeExpressionExecutors.length) {
+                boolean isExpectedParameterOverload = true;
+                for (int i = 0; i < overloadParameterNames.length; i++) {
+                    String overloadParameterName = overloadParameterNames[i];
+                    Parameter parameter = parameterMap.get(overloadParameterName);
+                    boolean supportedReturnType = false;
+                    for (DataType type : parameter.type()) {
+                        if (attributeExpressionExecutors[i].getReturnType().toString().
+                                equalsIgnoreCase(type.toString())) {
+                            supportedReturnType = true;
+                            break;
+                        }
+                    }
+                    if (!supportedReturnType) {
+                        isExpectedParameterOverload = false;
+                        break;
+                    }
+                }
+                if (isExpectedParameterOverload) {
+                    parameterOverload = aParameterOverload;
+                    break;
+                }
+            }
+        }
+
+        if (parameterOverload == null) {
+            if (parameterOverloads.length > 0) {
+                List<Attribute.Type> returnTypes = new ArrayList<>();
+                for (ExpressionExecutor expressionExecutor : attributeExpressionExecutors) {
+                    returnTypes.add(expressionExecutor.getReturnType());
+                }
+                throw new SiddhiAppValidationException("There is no parameterOverload for '" + key +
+                        "' that matches attribute types '" + returnTypes + "'.");
```

Do we need to throw this exception?

---

When parameterOverloads are given in the extension, the user should only use one of the given overloads. 
In this case, we can't find a matching overload and hence throwing an exception.